cmd/kubeadm/app/preflight/checks.go

```diff
@@ -819,8 +820,9 @@ func getEtcdVersionResponse(client *http.Client, url string, target interface{})
 
 // ImagePullCheck will pull container images used by kubeadm
 type ImagePullCheck struct {
-	runtime   utilruntime.ContainerRuntime
-	imageList []string
+	runtime          utilruntime.ContainerRuntime
+	imageList        []string
+	NodeRegistration kubeadmapi.NodeRegistrationOptions
```

Comment:
```suggestion
	imagePullPolicy v1.PullPolicy
```

instead of adding the whole NodeRegistration object we are only interested in the pull policy.